packages/makeswift-next-plugin/index.js

```diff
@@ -56,7 +65,115 @@ module.exports = ({ resolveSymlinks } = {}) => (nextConfig = {}) => {
     },
   }
 
-  return withTmInitializer(NEXT_TRANSPILE_MODULES_MODULES, {
-    resolveSymlinks,
-  })(enhancedConfig)
+  await createComponentsManifest()
+
+  return withTmInitializer(NEXT_TRANSPILE_MODULES_MODULES, { resolveSymlinks })(enhancedConfig)
+}
+
+async function createComponentsManifest() {
+  let makeswiftConfig
+  try {
+    // @todo: When will this assumption fail? Monorepos?
```

Comment:
Yeah, probably. We likely want a more robust way to find the config file

Reply:
We just need a reliable way to find the project root, i.e., where the `package.json` and `.next` folder are located. It's possible `process.cwd()` will always be the right one if running `next dev` or `next start` makes that always the case, but I would assume not. In a monorepo, doing `yarn workspace next-app dev` would be fine since Yarn workspaces sets the current working directory to the the workspace. The only case I can think of is the binary being executed from `node_modules` like `./node_modules/.bin/next dev`, but in that scenario, my guess is `next dev` (or `next start`) would require `process.cwd()` to be the directory with the `package.json` anyway so `process.cwd()` might be the right assumption...

Reply:
Ok cool. I've centralized this logic so we can isolate the assumption and have an easy place to change it if it's wrong. Once we settle everything else I can test some of these different scenarios further